visual-dotnet/SauceLabs.Visual/BuildFactory.cs

```diff
@@ -8,40 +9,44 @@ namespace SauceLabs.Visual
 {
     internal static class BuildFactory
     {
-        private static readonly Dictionary<string, VisualBuild> Builds = new Dictionary<string, VisualBuild>();
+        private static readonly Dictionary<string, ApiBuildPair> Builds = new Dictionary<string, ApiBuildPair>();
 
         internal static async Task<VisualBuild> Get(VisualApi api, CreateBuildOptions options)
         {
             // Check if there is already a build for the current region.
             if (Builds.TryGetValue(api.Region.Name, out var build))
             {
-                return build;
+                return build.Build;
             }
 
             var createdBuild = await Create(api, options);
-            Builds[api.Region.Name] = createdBuild;
+            Builds[api.Region.Name] = new ApiBuildPair(api.Clone(), createdBuild);
             return createdBuild;
         }
 
+        private static string? FindRegionByBuild(VisualBuild build)
+        {
+            return (from entry in Builds where entry.Value.Build == build select entry.Key).FirstOrDefault();
+        }
+
+        internal static async Task Close(VisualBuild build)
+        {
+            var key = FindRegionByBuild(build);
+            if (key != null)
+            {
+                var entry = Builds[key];
+                await entry.Api.FinishBuild(entry.Build.Id);
+                Builds.Remove(key);
+            }
```

Comment:
one less indent and check via recommended method
```suggestion
            if (string.IsNullOrEmpty(key)
            {
                return;
            }
            var entry = Builds[key];
            await entry.Api.FinishBuild(entry.Build.Id);
            Builds.Remove(key);
```